ingest/expression_files/mtx.py

```diff
@@ -191,14 +191,20 @@ def get_data_start_line_number(file_handler: IO) -> int:
          ----------
             i (IO): Line number where data starts
         """
-        i = 0
-        for line in file_handler:
-            if line.startswith("%"):
-                i += 1
-            else:
-                # First line w/o '%' is mtx dimension. So skip this line (+1)
-                i += 2
-                return i
+        for count, line in enumerate(file_handler):
+            if not line.startswith("%"):
+                try:
+                    line_values = line.strip().split()
+                    float(line_values[0])
+                    # First line w/o '%' is mtx dimension. So skip this line (+1)
+                    count += 2
```

Comment:
you shouldn't modify the count variable inside a loop like this--modifying iterators inside a loop is generally dangerous and confusing b/c it suggests that you've actually moved the file pointer.  just do `return count  + 2`

Reply:
Eno made this change in PR #175